near-sdk/src/collections/tree_map.rs

```diff
@@ -0,0 +1,1300 @@
+                self.ht.remove(&n);
+
+                if self.rgt.get(&p).map(|id| id == n).unwrap_or_default() {
+                    // n is on right link of p
+                    match self.lft.get(&n) {
+                        Some(l) => {
+                            self.rgt.insert(&p, &l);
+                            self.lft.remove(&n);
+                        },
+                        None => {
+                            self.rgt.remove(&p);
+                        }
+                    };
+                } else {
+                    // n is on left link of p
+                    match self.lft.get(&n) {
+                        Some(l) => {
+                            self.lft.insert(&p, &l);
+                            self.lft.remove(&n);
+                        },
+                        None => {
+                            self.lft.remove(&p);
+                        }
+                    }
+                }
+
+                // removing node might have caused a imbalance - balance the tree up to the root,
+                // starting from the lowest affected key (max key from left subtree in this case)
+                self.check_balance(root, &k)
+
+            } else if (b < 0 && rgt_opt.is_some()) || lft_opt.is_none() {
+                // proceed with right subtree
+                let rgt = rgt_opt.unwrap();
+
+                // k - min key from right subtree
+                let k = self.min_at(rgt).unwrap();
+
+                // n - id of a node that holds key k, p - id of an immediate parent of n
+                let (n, p) = self.lookup_at(r_id, &k).unwrap();
+
+                self.key.insert(&r_id, &k);
+                self.key.remove(&n);
+                self.ht.remove(&n);
+
+                if self.lft.get(&p).map(|id| id == n).unwrap_or_default() {
+                    // n is on left link of p
+                    match self.rgt.get(&n) {
+                        Some(r) => {
+                            self.lft.insert(&p, &r);
+                            self.rgt.remove(&n);
+                        },
+                        None => {
+                            self.lft.remove(&p);
+                        }
+                    }
+                } else {
+                    // n is on right link of p
+                    match self.rgt.get(&n) {
+                        Some(r) => {
+                            self.rgt.insert(&p, &r);
+                            self.rgt.remove(&n);
+                        },
+                        None => {
+                            self.rgt.remove(&p);
+                        }
+                    }
+                }
+
+                // removing node might have caused a imbalance - balance the tree up to the root,
+                // starting from the lowest affected key (min key from right subtree in this case)
+                self.check_balance(root, &k)
+
+            } else {
+                root
```

Comment:
This must be unreachable, right?
And then the previous `else if` can become `else`

Reply:
done